src/sdk.rs

```diff
@@ -257,3 +282,36 @@ pub fn panic_hex(data: &[u8]) -> ! {
     unsafe { exports::panic_utf8(message.len() as _, message.as_ptr() as _) }
     unreachable!()
 }
+
+#[inline]
+pub fn attached_deposit() -> u128 {
+    unsafe {
+        let result: u128 = 0;
+        exports::attached_deposit(result.as_ne_bytes().as_ptr() as _);
+        result
+    }
+}
+
+#[inline]
+pub fn storage_usage() -> u64 {
+    unsafe { exports::storage_usage() }
+}
+
+#[inline]
+pub fn storage_byte_cost() -> u128 {
+    // TODO: should be queried from host when available.
+    100_000_000_000_000_000_000
+}
+
+pub fn self_deploy(code: &[u8]) {
+    unsafe {
+        let account_id = current_account_id();
+        let promise_id =
+            exports::promise_batch_create(account_id.len() as _, account_id.as_ptr() as _);
+        exports::promise_batch_action_deploy_contract(
+            promise_id,
+            code.len() as _,
+            code.as_ptr() as _,
+        );
+    }
+}
```

Comment:
This section is a bit off. What is it trying to do?

Reply:
It deploys given code in place of the current code on this account via a promise (e.g. in the next block).

Reply:
This is following the "upgradable" pattern described here - https://github.com/near/NEPs/pull/123